RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -415,6 +423,22 @@ namespace {
     }
   }  // namespace dnn_inputs_2017_v2
 
+  float getTauID(const pat::Tau& tau, const std::string& tauID, float default_value = -999.) {
+    std::lock_guard<std::mutex> guard(g_mutex);
```

Comment:
- `guard` can be placed inside `else` section to not block execution for the cases when tau id is available.
- mutex can be defined as static inside `getTauID` instead of using the global namespace

Reply:
thanks!

Reply:
A better solution would be to use something like `tbb::concurrent_unordered_set<std::string>` for `isFirstWarning`, and check with something like
```c++
if (isFirstWarning.insert(tauID).second) {
   edm::LogWarning("TauID") << ...
}
```
which doesn't require any explicit locks.